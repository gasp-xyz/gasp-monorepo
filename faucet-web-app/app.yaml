runtime: nodejs18
service: react-faucet

handlers:
  - url: /static/js/(.*)
    static_files: build/static/js/\1
    upload: build/static/js/(.*)
    secure: always
  - url: /static/css/(.*)
    static_files: build/static/css/\1
    upload: build/static/css/(.*)
    secure: always
  - url: /static/media/(.*)
    static_files: build/static/media/\1
    upload: build/static/media/(.*)
    secure: always
  - url: /(.*\.(json|ico|png|jpg|svg))$
    static_files: build/\1
    upload: build/.*\.(json|ico|png|jpg|svg)$
    secure: always
  - url: /
    static_files: build/index.html
    upload: build/index.html
    secure: always
  - url: /.*
    static_files: build/index.html
    upload: build/index.html
    secure: always