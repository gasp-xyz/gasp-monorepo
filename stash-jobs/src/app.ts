import 'gasp-types'
import express, { Request, Response, json, urlencoded } from 'express'
import cors from 'cors'

import { createRequire } from 'module'
const require = createRequire(import.meta.url)
const swaggerFile = require('../swagger-output.json')

const app = express()

// App configuration
app.set('port', process.env.PORT || 8080)
app.use(json())
app.use(urlencoded({ extended: true }))
app.use(cors())

//Root (Health Check)
app.get('/', async (req: Request, res: Response): Promise<void> => {
  res.json({})
})
export default app
