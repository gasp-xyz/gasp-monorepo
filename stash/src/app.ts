import 'gasp-types'
import express, { Request, Response, json, urlencoded } from 'express'
import swaggerUI from 'swagger-ui-express'
import cors from 'cors'

import * as priceDiscoveryController from './controller/PriceDiscoveryController.js'
import * as tokenController from './controller/TokenController.js'
import * as xcmController from './controller/XcmController.js'
import * as coingeckoController from './controller/coingeckoListingController.js'
import * as coinmarketcapController from './controller/coinmarketcapListingController.js'
import * as tokenListController from './controller/tokenListController.js'
import * as stakingAprController from './controller/stakingAprController.js'
import * as liquidStakingController from './controller/liquidStakingController.js'
import * as networkController from './controller/networkController.js'
import * as tokenNetworkPortfolioController from './controller/tokenNetworkPortfolioController.js'
import * as faucetController from './controller/FaucetController.js'
import * as tracingController from './controller/TracingController.js'

import { createRequire } from 'module'
const require = createRequire(import.meta.url)
const swaggerFile = require('../swagger-output.json')

const app = express()

// App configuration
app.set('port', process.env.PORT || 8080)
app.use(json())
app.use(urlencoded({ extended: true }))
app.use(cors())

//Root (Health Check)
app.get('/', async (req: Request, res: Response): Promise<void> => {
  res.json({})
})

// Tokens
app.get('/token/order-buckets', tokenController.listTokenOrderBuckets)
// Token list
app.get('/token/list/stats', tokenListController.tokenList)
app.get('/token/:id/stats', tokenListController.tokenDetails)

// XCM
app.get('/xcm/channels', xcmController.channels)
app.get('/xcm/network', xcmController.network)
app.get('/xcm/tokens', xcmController.tokens)

app.get(
  '/collator/:collatorAddress/staking/dailyReward',
  stakingAprController.dailyReward
)
app.get('/collator/:collatorAddress/staking/apy', stakingAprController.apy)
app.get('/collators/staking/apy', stakingAprController.collatorsApy)

app.get(
  '/account/:address/liquid-staking/rewards-history/24h/sum',
  liquidStakingController.rewards24hours
)
app.get(
  '/account/:address/liquid-staking/rewards-history/month/sum',
  liquidStakingController.rewardsMonth
)

// Price Discovery API
app.get('/price-discovery/:currencyId/', priceDiscoveryController.getPrice)
app.get(
  '/price-history/pair/:baseCurrencyId/:targetCurrencyId',
  priceDiscoveryController.getHistoryPair
)
app.get('/price-history/:currencyId/', priceDiscoveryController.getHistory)
app.get(
  '/volume-history/pools/:currencyId/',
  priceDiscoveryController.getTradesPool
)
app.get('/volume-history/:currencyId/', priceDiscoveryController.getTradesAsset)
app.get(
  '/tvl-history/pools/:currencyId/',
  priceDiscoveryController.getVolumePool
)
app.get('/tvl-history/:currencyId/', priceDiscoveryController.getVolumeAsset)

// CoinGecko listing endpoints
app.get('/coingecko/pairs', coingeckoController.pairs)
app.get('/coingecko/tickers', coingeckoController.tickers)

app.get('/affirmed-network/list', networkController.networkList)
app.get('/affirmed-token/list', networkController.tokenList)
app.get(
  '/account/:address/token-portfolio',
  tokenNetworkPortfolioController.tokenNetworkPortfolio
)

//Faucet endpoint
app.get(
  '/faucet/requestTokens/:toAddress/captcha/:captcha/',
  faucetController.captcha
)

// Tracing endpoints
app.post('/tracing/tx/start', tracingController.startTracing)

app.get(
  '/tracing/tx/:txHashOrEntityId',
  tracingController.getTransactionStatusByTxHashOrEntityId
)

app.get(
  '/tracing/tx/listByAddress/:address',
  tracingController.getAllTransactionsByAddress
)

app.get(
  '/tracing/tx/listByAddress/:address/:status',
  tracingController.getAllTransactionsByAddressAndStatus
)

app.get(
  '/tracing/tx/findByEntityId/:entityId',

  tracingController.getATransactionByEntityId
)

// Coinmarketcap listing endpoints
app.get('/coinmarketcap/v1/summary', coinmarketcapController.summary)

app.use('/doc', swaggerUI.serve, swaggerUI.setup(swaggerFile))

export default app
