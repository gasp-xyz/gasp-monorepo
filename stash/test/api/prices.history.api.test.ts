
import { describe, it , expect} from 'vitest'
import app from "../../src/app";
import supertest from "supertest";

const priceHistoryPath = "price-history"
describe('Prices', () => {

  describe('/GET prices', () => {
    it('[MGX-597] - pools should not be returned on prices', async() => {
      await supertest(app)
        .get(`/${priceHistoryPath}/GASPV2-L1Asset?interval=day&days=300`)
        .expect(500)
        .then((response) => {
          const invalidTokenNameResponse = response.body;
          expect(invalidTokenNameResponse.exceptionName).to.contain("ValidationError")
          expect(invalidTokenNameResponse.message).to.contain("this must be one of the following values: GASPV2, L1Asset, GASPV2-ETH, L1Asset-GASPV2")
        });
    });

    it('GET /price-history - Schema validation', async() => {
      await supertest(app)
        .get(`/${priceHistoryPath}/GASPV2`)         
        .expect(200)
        .then((response) => {
          expect(response.body).toMatchSnapshot("Prices")
          expect(response.body.error).toBeUndefined();
          expect(response.body.prices).toBeDefined();
          expect(response.body.prices).toBeInstanceOf(Array);
        });
    });
  });
});
