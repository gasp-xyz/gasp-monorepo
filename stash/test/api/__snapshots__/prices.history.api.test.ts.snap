// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Prices > /GET prices > GET /price-history - Schema validation > Prices 1`] = `
{
  "prices": [
    [
      1680307200000,
      "0.0011472158206295857",
    ],
    [
      1680393600000,
      "0.0011161007912752026",
    ],
    [
      1680480000000,
      "0.0010879391097295395",
    ],
    [
      1680566400000,
      "0.0010914175450533088",
    ],
    [
      1680652800000,
      "0.0011064981996454728",
    ],
    [
      1680739200000,
      "0.0010928161274552437",
    ],
    [
      1680825600000,
      "0.001069149143442327",
    ],
    [
      1680912000000,
      "0.0010625325310309022",
    ],
    [
      1680998400000,
      "0.0010597097810295178",
    ],
    [
      1681084800000,
      "0.0010536599297755227",
    ],
    [
      1681171200000,
      "0.0010664147500974006",
    ],
    [
      1681257600000,
      "0.0010840150607882656",
    ],
    [
      1681344000000,
      "0.0011036254737166818",
    ],
    [
      1681430400000,
      "0.0011438567728187189",
    ],
    [
      1681516800000,
      "0.0012137605147875648",
    ],
    [
      1681603200000,
      "0.001229550387092294",
    ],
    [
      1681689600000,
      "0.0012621579333899234",
    ],
    [
      1681776000000,
      "0.0012069483932569319",
    ],
    [
      1681862400000,
      "0.0012447312020409813",
    ],
    [
      1681948800000,
      "0.0011200888875994217",
    ],
    [
      1682035200000,
      "0.0010709806879939692",
    ],
    [
      1682121600000,
      "0.0010172922340505059",
    ],
    [
      1682208000000,
      "0.0010450991792025094",
    ],
    [
      1682294400000,
      "0.001018675674553883",
    ],
    [
      1682380800000,
      "0.0010160565709638459",
    ],
    [
      1682467200000,
      "0.0010431360450847838",
    ],
    [
      1682553600000,
      "0.0010260618072088418",
    ],
    [
      1682640000000,
      "0.0010536972871840245",
    ],
    [
      1682726400000,
      "0.0010520253197199372",
    ],
    [
      1682812800000,
      "0.0010773959002226444",
    ],
    [
      1682899200000,
      "0.0010679251500398482",
    ],
    [
      1682985600000,
      "0.000993124792849398",
    ],
    [
      1683072000000,
      "0.0009966835085340832",
    ],
    [
      1683158400000,
      "0.0009921435987197815",
    ],
    [
      1683244800000,
      "0.0009669392365936701",
    ],
    [
      1683331200000,
      "0.0009683986173303022",
    ],
    [
      1683417600000,
      "0.0009232933367819139",
    ],
    [
      1683504000000,
      "0.0009234743034448038",
    ],
    [
      1683590400000,
      "0.0008906167106422688",
    ],
    [
      1683676800000,
      "0.0008910306765842568",
    ],
    [
      1683763200000,
      "0.0009041855219285646",
    ],
    [
      1683849600000,
      "0.00086546637406733",
    ],
    [
      1683936000000,
      "0.000867426946630889",
    ],
    [
      1684022400000,
      "0.0008505121509288051",
    ],
    [
      1684108800000,
      "0.0008685984814193842",
    ],
    [
      1684195200000,
      "0.0008750337191851281",
    ],
    [
      1684281600000,
      "0.0008804618936618678",
    ],
    [
      1684368000000,
      "0.0008888653546646935",
    ],
    [
      1684454400000,
      "0.0008611419693239368",
    ],
    [
      1684540800000,
      "0.0008426877126791799",
    ],
    [
      1684627200000,
      "0.000853245620611938",
    ],
    [
      1684713600000,
      "0.0008314981454787805",
    ],
    [
      1684800000000,
      "0.0008198518705837154",
    ],
    [
      1684886400000,
      "0.0008209803487973372",
    ],
    [
      1684972800000,
      "0.0008064805627311667",
    ],
    [
      1685059200000,
      "0.0008080953326006976",
    ],
    [
      1685145600000,
      "0.0008038148146948566",
    ],
    [
      1685232000000,
      "0.0008143642514566597",
    ],
    [
      1685318400000,
      "0.0008296177355787072",
    ],
    [
      1685404800000,
      "0.0008218220417823708",
    ],
    [
      1685491200000,
      "0.0008195359441942263",
    ],
    [
      1685577600000,
      "0.00082625190926407325",
    ],
    [
      1685664000000,
      "0.0008134241430865131",
    ],
    [
      1685750400000,
      "0.0008186253961584328",
    ],
    [
      1685836800000,
      "0.0008343718287627492",
    ],
    [
      1685923200000,
      "0.000862790989803326",
    ],
    [
      1686009600000,
      "0.0008012151377243624",
    ],
    [
      1686096000000,
      "0.0008430993720092495",
    ],
    [
      1686182400000,
      "0.0008097040198050509",
    ],
    [
      1686268800000,
      "0.0007928655204075253",
    ],
    [
      1686355200000,
      "0.0007640828199105108",
    ],
    [
      1686441600000,
      "0.000658479472148573",
    ],
    [
      1686528000000,
      "0.0006604420251754108",
    ],
    [
      1686614400000,
      "0.0006575195207243178",
    ],
    [
      1686700800000,
      "0.0006293676995169258",
    ],
    [
      1686787200000,
      "0.0005957339595099796",
    ],
    [
      1686873600000,
      "0.000587611192144739",
    ],
    [
      1686960000000,
      "0.0006174890976222831",
    ],
    [
      1687046400000,
      "0.0006384205575838871",
    ],
    [
      1687132800000,
      "0.0006327830292174725",
    ],
    [
      1687219200000,
      "0.0006414037711741031",
    ],
    [
      1687305600000,
      "0.0006591751229027964",
    ],
    [
      1687392000000,
      "0.0006674986062644386",
    ],
    [
      1687478400000,
      "0.0006525129126182298",
    ],
    [
      1687564800000,
      "0.0006722530162388232",
    ],
    [
      1687651200000,
      "0.0006729777358184478",
    ],
    [
      1687737600000,
      "0.0007106784395501534",
    ],
    [
      1687824000000,
      "0.0006702886750173839",
    ],
    [
      1687910400000,
      "0.0007060286084988504",
    ],
    [
      1687996800000,
      "0.00067319554745830615",
    ],
    [
      1688083200000,
      "0.0006872348753568107",
    ],
    [
      1688169600000,
      "0.0006887482113989275",
    ],
    [
      1688256000000,
      "0.0007151525444295799",
    ],
    [
      1688342400000,
      "0.0007262261711873494",
    ],
    [
      1688428800000,
      "0.0007213243904909606",
    ],
    [
      1688515200000,
      "0.0006916172171093924",
    ],
    [
      1688601600000,
      "0.0006473029777629985",
    ],
    [
      1688688000000,
      "0.000625710548937773",
    ],
    [
      1688774400000,
      "0.0006308629631755669",
    ],
    [
      1688860800000,
      "0.00062393131929810725",
    ],
    [
      1689206400000,
      "0.0006197309087928698",
    ],
    [
      1689292800000,
      "0.0006576731972092575",
    ],
    [
      1689379200000,
      "0.0006353693602171516",
    ],
    [
      1689465600000,
      "0.0006322425494322246",
    ],
    [
      1689552000000,
      "0.0006099294104461807",
    ],
    [
      1689638400000,
      "0.0006241027410013117",
    ],
    [
      1689724800000,
      "0.0006129174824501794",
    ],
    [
      1689811200000,
      "0.0006067794727437284",
    ],
    [
      1689897600000,
      "0.0005972823217118889",
    ],
    [
      1689984000000,
      "0.0006146691073025223",
    ],
    [
      1690070400000,
      "0.000601027651047617",
    ],
    [
      1690156800000,
      "0.0006118881991575841",
    ],
    [
      1690243200000,
      "0.0005876482637802151",
    ],
    [
      1690329600000,
      "0.0005812905803274245",
    ],
    [
      1690416000000,
      "0.0005831508441759465",
    ],
    [
      1690502400000,
      "0.0005814124433749934",
    ],
    [
      1690588800000,
      "0.0005949100273912282",
    ],
    [
      1690675200000,
      "0.0005925514586931284",
    ],
    [
      1690761600000,
      "0.0005769915143218424",
    ],
    [
      1690848000000,
      "0.0005589397431531062",
    ],
    [
      1690934400000,
      "0.000580963871052058",
    ],
    [
      1691020800000,
      "0.0005731521297066728",
    ],
    [
      1691107200000,
      "0.000566362303912052",
    ],
    [
      1691193600000,
      "0.0005440619880725058",
    ],
    [
      1691280000000,
      "0.0005241816820738766",
    ],
    [
      1691366400000,
      "0.0005306246672345021",
    ],
    [
      1691452800000,
      "0.0005386103277692302",
    ],
    [
      1691539200000,
      "0.0005475632148147478",
    ],
    [
      1691625600000,
      "0.0005587169178227292",
    ],
    [
      1691712000000,
      "0.0005517897131362994",
    ],
    [
      1691798400000,
      "0.0005414681836721549",
    ],
    [
      1691884800000,
      "0.0005726505980967787",
    ],
    [
      1691971200000,
      "0.0005768841885888463",
    ],
    [
      1692057600000,
      "0.0005768634857261362",
    ],
    [
      1692144000000,
      "0.0005590035850185792",
    ],
    [
      1692230400000,
      "0.0005521701167002144",
    ],
    [
      1692316800000,
      "0.0005164284877865564",
    ],
    [
      1692403200000,
      "0.0005115717848226708",
    ],
    [
      1692489600000,
      "0.0005154395335222643",
    ],
    [
      1692576000000,
      "0.0005162640543085188",
    ],
    [
      1692662400000,
      "0.0004973088818646634",
    ],
    [
      1692748800000,
      "0.0004851077481153163",
    ],
    [
      1692835200000,
      "0.0005055930375498087",
    ],
    [
      1692921600000,
      "0.0004840717688783257",
    ],
    [
      1693008000000,
      "0.0004820919482658608",
    ],
    [
      1693094400000,
      "0.00048301773419401535",
    ],
    [
      1693180800000,
      "0.0004787467149205729",
    ],
    [
      1693267200000,
      "0.0004739138216843306",
    ],
    [
      1693353600000,
      "0.0004827054925435023",
    ],
    [
      1693440000000,
      "0.0004729520682198325",
    ],
    [
      1693526400000,
      "0.0004580615125206925",
    ],
    [
      1693612800000,
      "0.0004546365788233505",
    ],
    [
      1693699200000,
      "0.0004602450241085659",
    ],
    [
      1693785600000,
      "0.0004553750625595522",
    ],
    [
      1693872000000,
      "0.0004545031639919847",
    ],
    [
      1693958400000,
      "0.0004617145015878763",
    ],
    [
      1694044800000,
      "0.0004577395658631211",
    ],
    [
      1694131200000,
      "0.0004643821548713696",
    ],
    [
      1694217600000,
      "0.0004599974010734144",
    ],
    [
      1694304000000,
      "0.0004619108431438293",
    ],
    [
      1694390400000,
      "0.0004475066635939499",
    ],
    [
      1694476800000,
      "0.0004299591148592601",
    ],
    [
      1694563200000,
      "0.0004316435667485248",
    ],
    [
      1694649600000,
      "0.0004351125100047613",
    ],
    [
      1694736000000,
      "0.000445082578327747",
    ],
    [
      1694822400000,
      "0.0004644762344824017",
    ],
    [
      1694908800000,
      "0.0004703188102058208",
    ],
    [
      1694995200000,
      "0.0004543280924026659",
    ],
    [
      1695081600000,
      "0.0004628649889969971",
    ],
    [
      1695168000000,
      "0.0004658403522913232",
    ],
    [
      1695254400000,
      "0.0004676354925513806",
    ],
    [
      1695340800000,
      "0.00045799485428133085",
    ],
    [
      1695427200000,
      "0.0004496223290216384",
    ],
    [
      1695513600000,
      "0.0004466528607371759",
    ],
    [
      1695600000000,
      "0.0004368240512379402",
    ],
    [
      1695686400000,
      "0.0004356385942859617",
    ],
    [
      1695772800000,
      "0.0004211828675981167",
    ],
    [
      1695859200000,
      "0.0004205642189464598",
    ],
    [
      1695945600000,
      "0.0004188450998719874",
    ],
    [
      1699488000000,
      "0.0004635584097289272",
    ],
    [
      1699574400000,
      "0.0004313927602671381",
    ],
    [
      1699660800000,
      "0.0004518447135196338",
    ],
    [
      1699747200000,
      "0.000453919023182876",
    ],
    [
      1699833600000,
      "0.000475793510681639",
    ],
  ],
}
`;

exports[`Prices > /GET prices > should return something with GASPV2 > Prices 1`] = `
{
  "prices": [
    [
      1680307200000,
      "0.0011472158206295857",
    ],
    [
      1680393600000,
      "0.0011161007912752026",
    ],
    [
      1680480000000,
      "0.0010879391097295395",
    ],
    [
      1680566400000,
      "0.0010914175450533088",
    ],
    [
      1680652800000,
      "0.0011064981996454728",
    ],
    [
      1680739200000,
      "0.0010928161274552437",
    ],
    [
      1680825600000,
      "0.001069149143442327",
    ],
    [
      1680912000000,
      "0.0010625325310309022",
    ],
    [
      1680998400000,
      "0.0010597097810295178",
    ],
    [
      1681084800000,
      "0.0010536599297755227",
    ],
    [
      1681171200000,
      "0.0010664147500974006",
    ],
    [
      1681257600000,
      "0.0010840150607882656",
    ],
    [
      1681344000000,
      "0.0011036254737166818",
    ],
    [
      1681430400000,
      "0.0011438567728187189",
    ],
    [
      1681516800000,
      "0.0012137605147875648",
    ],
    [
      1681603200000,
      "0.001229550387092294",
    ],
    [
      1681689600000,
      "0.0012621579333899234",
    ],
    [
      1681776000000,
      "0.0012069483932569319",
    ],
    [
      1681862400000,
      "0.0012447312020409813",
    ],
    [
      1681948800000,
      "0.0011200888875994217",
    ],
    [
      1682035200000,
      "0.0010709806879939692",
    ],
    [
      1682121600000,
      "0.0010172922340505059",
    ],
    [
      1682208000000,
      "0.0010450991792025094",
    ],
    [
      1682294400000,
      "0.001018675674553883",
    ],
    [
      1682380800000,
      "0.0010160565709638459",
    ],
    [
      1682467200000,
      "0.0010431360450847838",
    ],
    [
      1682553600000,
      "0.0010260618072088418",
    ],
    [
      1682640000000,
      "0.0010536972871840245",
    ],
    [
      1682726400000,
      "0.0010520253197199372",
    ],
    [
      1682812800000,
      "0.0010773959002226444",
    ],
    [
      1682899200000,
      "0.0010679251500398482",
    ],
    [
      1682985600000,
      "0.000993124792849398",
    ],
    [
      1683072000000,
      "0.0009966835085340832",
    ],
    [
      1683158400000,
      "0.0009921435987197815",
    ],
    [
      1683244800000,
      "0.0009669392365936701",
    ],
    [
      1683331200000,
      "0.0009683986173303022",
    ],
    [
      1683417600000,
      "0.0009232933367819139",
    ],
    [
      1683504000000,
      "0.0009234743034448038",
    ],
    [
      1683590400000,
      "0.0008906167106422688",
    ],
    [
      1683676800000,
      "0.0008910306765842568",
    ],
    [
      1683763200000,
      "0.0009041855219285646",
    ],
    [
      1683849600000,
      "0.00086546637406733",
    ],
    [
      1683936000000,
      "0.000867426946630889",
    ],
    [
      1684022400000,
      "0.0008505121509288051",
    ],
    [
      1684108800000,
      "0.0008685984814193842",
    ],
    [
      1684195200000,
      "0.0008750337191851281",
    ],
    [
      1684281600000,
      "0.0008804618936618678",
    ],
    [
      1684368000000,
      "0.0008888653546646935",
    ],
    [
      1684454400000,
      "0.0008611419693239368",
    ],
    [
      1684540800000,
      "0.0008426877126791799",
    ],
    [
      1684627200000,
      "0.000853245620611938",
    ],
    [
      1684713600000,
      "0.0008314981454787805",
    ],
    [
      1684800000000,
      "0.0008198518705837154",
    ],
    [
      1684886400000,
      "0.0008209803487973372",
    ],
    [
      1684972800000,
      "0.0008064805627311667",
    ],
    [
      1685059200000,
      "0.0008080953326006976",
    ],
    [
      1685145600000,
      "0.0008038148146948566",
    ],
    [
      1685232000000,
      "0.0008143642514566597",
    ],
    [
      1685318400000,
      "0.0008296177355787072",
    ],
    [
      1685404800000,
      "0.0008218220417823708",
    ],
    [
      1685491200000,
      "0.0008195359441942263",
    ],
    [
      1685577600000,
      "0.00082625190926407325",
    ],
    [
      1685664000000,
      "0.0008134241430865131",
    ],
    [
      1685750400000,
      "0.0008186253961584328",
    ],
    [
      1685836800000,
      "0.0008343718287627492",
    ],
    [
      1685923200000,
      "0.000862790989803326",
    ],
    [
      1686009600000,
      "0.0008012151377243624",
    ],
    [
      1686096000000,
      "0.0008430993720092495",
    ],
    [
      1686182400000,
      "0.0008097040198050509",
    ],
    [
      1686268800000,
      "0.0007928655204075253",
    ],
    [
      1686355200000,
      "0.0007640828199105108",
    ],
    [
      1686441600000,
      "0.000658479472148573",
    ],
    [
      1686528000000,
      "0.0006604420251754108",
    ],
    [
      1686614400000,
      "0.0006575195207243178",
    ],
    [
      1686700800000,
      "0.0006293676995169258",
    ],
    [
      1686787200000,
      "0.0005957339595099796",
    ],
    [
      1686873600000,
      "0.000587611192144739",
    ],
    [
      1686960000000,
      "0.0006174890976222831",
    ],
    [
      1687046400000,
      "0.0006384205575838871",
    ],
    [
      1687132800000,
      "0.0006327830292174725",
    ],
    [
      1687219200000,
      "0.0006414037711741031",
    ],
    [
      1687305600000,
      "0.0006591751229027964",
    ],
    [
      1687392000000,
      "0.0006674986062644386",
    ],
    [
      1687478400000,
      "0.0006525129126182298",
    ],
    [
      1687564800000,
      "0.0006722530162388232",
    ],
    [
      1687651200000,
      "0.0006729777358184478",
    ],
    [
      1687737600000,
      "0.0007106784395501534",
    ],
    [
      1687824000000,
      "0.0006702886750173839",
    ],
    [
      1687910400000,
      "0.0007060286084988504",
    ],
    [
      1687996800000,
      "0.00067319554745830615",
    ],
    [
      1688083200000,
      "0.0006872348753568107",
    ],
    [
      1688169600000,
      "0.0006887482113989275",
    ],
    [
      1688256000000,
      "0.0007151525444295799",
    ],
    [
      1688342400000,
      "0.0007262261711873494",
    ],
    [
      1688428800000,
      "0.0007213243904909606",
    ],
    [
      1688515200000,
      "0.0006916172171093924",
    ],
    [
      1688601600000,
      "0.0006473029777629985",
    ],
    [
      1688688000000,
      "0.000625710548937773",
    ],
    [
      1688774400000,
      "0.0006308629631755669",
    ],
    [
      1688860800000,
      "0.00062393131929810725",
    ],
    [
      1689206400000,
      "0.0006197309087928698",
    ],
    [
      1689292800000,
      "0.0006576731972092575",
    ],
    [
      1689379200000,
      "0.0006353693602171516",
    ],
    [
      1689465600000,
      "0.0006322425494322246",
    ],
    [
      1689552000000,
      "0.0006099294104461807",
    ],
    [
      1689638400000,
      "0.0006241027410013117",
    ],
    [
      1689724800000,
      "0.0006129174824501794",
    ],
    [
      1689811200000,
      "0.0006067794727437284",
    ],
    [
      1689897600000,
      "0.0005972823217118889",
    ],
    [
      1689984000000,
      "0.0006146691073025223",
    ],
    [
      1690070400000,
      "0.000601027651047617",
    ],
    [
      1690156800000,
      "0.0006118881991575841",
    ],
    [
      1690243200000,
      "0.0005876482637802151",
    ],
    [
      1690329600000,
      "0.0005812905803274245",
    ],
    [
      1690416000000,
      "0.0005831508441759465",
    ],
    [
      1690502400000,
      "0.0005814124433749934",
    ],
    [
      1690588800000,
      "0.0005949100273912282",
    ],
    [
      1690675200000,
      "0.0005925514586931284",
    ],
    [
      1690761600000,
      "0.0005769915143218424",
    ],
    [
      1690848000000,
      "0.0005589397431531062",
    ],
    [
      1690934400000,
      "0.000580963871052058",
    ],
    [
      1691020800000,
      "0.0005731521297066728",
    ],
    [
      1691107200000,
      "0.000566362303912052",
    ],
    [
      1691193600000,
      "0.0005440619880725058",
    ],
    [
      1691280000000,
      "0.0005241816820738766",
    ],
    [
      1691366400000,
      "0.0005306246672345021",
    ],
    [
      1691452800000,
      "0.0005386103277692302",
    ],
    [
      1691539200000,
      "0.0005475632148147478",
    ],
    [
      1691625600000,
      "0.0005587169178227292",
    ],
    [
      1691712000000,
      "0.0005517897131362994",
    ],
    [
      1691798400000,
      "0.0005414681836721549",
    ],
    [
      1691884800000,
      "0.0005726505980967787",
    ],
    [
      1691971200000,
      "0.0005768841885888463",
    ],
    [
      1692057600000,
      "0.0005768634857261362",
    ],
    [
      1692144000000,
      "0.0005590035850185792",
    ],
    [
      1692230400000,
      "0.0005521701167002144",
    ],
    [
      1692316800000,
      "0.0005164284877865564",
    ],
    [
      1692403200000,
      "0.0005115717848226708",
    ],
    [
      1692489600000,
      "0.0005154395335222643",
    ],
    [
      1692576000000,
      "0.0005162640543085188",
    ],
    [
      1692662400000,
      "0.0004973088818646634",
    ],
    [
      1692748800000,
      "0.0004851077481153163",
    ],
    [
      1692835200000,
      "0.0005055930375498087",
    ],
    [
      1692921600000,
      "0.0004840717688783257",
    ],
    [
      1693008000000,
      "0.0004820919482658608",
    ],
    [
      1693094400000,
      "0.00048301773419401535",
    ],
    [
      1693180800000,
      "0.0004787467149205729",
    ],
    [
      1693267200000,
      "0.0004739138216843306",
    ],
    [
      1693353600000,
      "0.0004827054925435023",
    ],
    [
      1693440000000,
      "0.0004729520682198325",
    ],
    [
      1693526400000,
      "0.0004580615125206925",
    ],
    [
      1693612800000,
      "0.0004546365788233505",
    ],
    [
      1693699200000,
      "0.0004602450241085659",
    ],
    [
      1693785600000,
      "0.0004553750625595522",
    ],
    [
      1693872000000,
      "0.0004545031639919847",
    ],
    [
      1693958400000,
      "0.0004617145015878763",
    ],
    [
      1694044800000,
      "0.0004577395658631211",
    ],
    [
      1694131200000,
      "0.0004643821548713696",
    ],
    [
      1694217600000,
      "0.0004599974010734144",
    ],
    [
      1694304000000,
      "0.0004619108431438293",
    ],
    [
      1694390400000,
      "0.0004475066635939499",
    ],
    [
      1694476800000,
      "0.0004299591148592601",
    ],
    [
      1694563200000,
      "0.0004316435667485248",
    ],
    [
      1694649600000,
      "0.0004351125100047613",
    ],
    [
      1694736000000,
      "0.000445082578327747",
    ],
    [
      1694822400000,
      "0.0004644762344824017",
    ],
    [
      1694908800000,
      "0.0004703188102058208",
    ],
    [
      1694995200000,
      "0.0004543280924026659",
    ],
    [
      1695081600000,
      "0.0004628649889969971",
    ],
    [
      1695168000000,
      "0.0004658403522913232",
    ],
    [
      1695254400000,
      "0.0004676354925513806",
    ],
    [
      1695340800000,
      "0.00045799485428133085",
    ],
    [
      1695427200000,
      "0.0004496223290216384",
    ],
    [
      1695513600000,
      "0.0004466528607371759",
    ],
    [
      1695600000000,
      "0.0004368240512379402",
    ],
    [
      1695686400000,
      "0.0004356385942859617",
    ],
    [
      1695772800000,
      "0.0004211828675981167",
    ],
    [
      1695859200000,
      "0.0004205642189464598",
    ],
    [
      1695945600000,
      "0.0004188450998719874",
    ],
    [
      1699488000000,
      "0.0004635584097289272",
    ],
    [
      1699574400000,
      "0.0004313927602671381",
    ],
    [
      1699660800000,
      "0.0004518447135196338",
    ],
    [
      1699747200000,
      "0.000453919023182876",
    ],
    [
      1699833600000,
      "0.000475793510681639",
    ],
  ],
}
`;
